src/Controller/CRUDController.php

```diff
@@ -753,10 +733,8 @@ public function historyViewRevisionAction(Request $request, $revision = null)
      *
      * @throws AccessDeniedException If access is not granted
      * @throws NotFoundHttpException If the object or revision does not exist or the audit reader is not available
-     *
-     * @return Response
      */
-    public function historyCompareRevisionsAction(Request $request, $baseRevision = null, $compareRevision = null)
+    public function historyCompareRevisionsAction(Request $request, $baseRevision = null, $compareRevision = null): Response
```

Comment:
Similar to `historyViewRevisionAction`, isn't `baseRevision` and `compareRevision` always string ?

Reply:
True, updated now.  I've also updated tests, hope they are good, however I'm not sure about [L3115](https://github.com/sonata-project/SonataAdminBundle/blob/7fbe5f54d9ffe555f7b505f3b49b612b5d2b5473/tests/Controller/CRUDControllerTest.php#L3115) in
`testHistoryCompareRevisionsActionNotFoundCompareRevision`  and the second argument (at the moment it's set to `null`)

Reply:
```
$reader->expects($this->exactly(2))->method('find')->willReturnMap([
    ['Foo', 123, 'fooBaseRevision', $objectRevision],
    ['Foo', 123, 'fooCompareRevision', null],
]);
```
means
- find('Foo', 123, 'fooBaseRevision') = $objectRevision
- find('Foo', 123, 'fooCompareRevision') = null

So I'm not sure to understand your issue

Reply:
Thanks. I just wasn't sure what that line means and if the test still make sense after my changes or something more should be updated.